vitrina/datasets/views.py

```diff
@@ -14,3 +20,168 @@ class DatasetListView(ListView):
 class DatasetDetailView(DetailView):
     model = Dataset
     template_name = 'vitrina/datasets/detail.html'
+
+
+class DatasetSearchResultsView(ListView):
+    model = Dataset
+    template_name = 'vitrina/datasets/list.html'
+    paginate_by = 20
+
+    def get_queryset(self):
+        query = self.request.GET.get('q')
+        date_from = self.request.GET.get('date_from')
+        date_to = self.request.GET.get('date_to')
+        status = self.request.GET.get('status')
+        tags = self.request.GET.getlist('tags')
+        categories = self.request.GET.getlist('category')
+        filter_dict = self.request.GET.dict()
+        filter_dict.pop('q', None)
+        filter_dict.pop('date_from', None)
+        filter_dict.pop('date_to', None)
+        filter_dict.pop('status', None)
+        filter_dict.pop('tags', None)
+        filter_dict.pop('category', None)
+        datasets = Dataset.public.order_by('-published')
+
+        if query:
+            datasets = datasets.filter(
+                Q(title__icontains=query) | Q(title_en__icontains=query)
+            )
+        if date_from and date_to:
+            datasets = datasets.filter(Q(published__gt=date_from) & Q(published__lt=date_to))
```

Comment:
```suggestion
        if date_from and date_to:
            datasets = datasets.filter(published__range=(date_from, date_to))
```

https://docs.djangoproject.com/en/3.2/ref/models/querysets/#range

Filtering by dates should be inclusive.